lib/perl/Genome/Model/Tools/Analysis/Coverage/CoveragePlot.pm

```diff
@@ -0,0 +1,1375 @@
+        
+        my $c = $count{$filter}->{total_covered_bases} / $count{$filter}->{total_ref_bases} * 100;;
+        my $d = $count{$filter}->{sum_cov_depth} / $count{$filter}->{total_ref_bases};
+        
+        printf STDERR "\n%d\t%.2f\t%u\t%u\t%.2f", $filter, $c, $count{$filter}->{total_ref_bases}, $count{$filter}->{total_covered_bases}, $d;
+    }
+    
+    
+    # writes the boxplot statistics
+    $tabdist->write(catfile($dirout, "boxplot.tab"));
+    
+    
+    # sorts the table
+    $tabstat->{_data} = [ map { $_->[0] }
+                sort {
+                            $a->[1] cmp $b->[1]
+                                    ||
+                            $a->[2] <=> $b->[2]
+                       }
+                map { [$_, $_->[0], $_->[1]] } @{$tabstat->{_data}} ];
+    
+    # writes the tabular table in tab delimited text format
+    $tabstat->write(catfile($dirout, "coverage.tab"));
+    
+    
+    # sorts the table
+    $tabden->{_data} = [ map { $_->[0] }
+                sort {
+                            $a->[1] cmp $b->[1]
+                                    ||
+                            $a->[2] <=> $b->[2]
+                       }
+                map { [$_, $_->[0], $_->[1]] } @{$tabden->{_data}} ];
+    
+    # writes out the density function
+    $tabden->write(catfile($dirout, "density.tab"));
+    
+    
+    $R->stopR();
+    
+    
+    # the end of the main subroutine
+}
+
+
+sub get1rvar {
+    # gets the variable as a vector
+    my ($R) = shift;
+    my ($var) = @_;
+
+    # local variables
+    my @arr;
+    
+    # gets the read
+    $R->send('print(' . $var . ')');
+    my $print = $R->read;
+    
+    if (defined $print && $print =~ /^\s*\[1\]\s+/)
+    {
+        # replaces the multiple numbers
+        $print =~ s/\s*\[\d+\]\s*/ /g;
+        $print =~ s/^\s+//g;
+        $print =~ s/\s+$//g;
+        
+        # gets the value(s)
+        @arr = split /\s+/, $print;
+    }
+    else
+    {
+        # NULL
+        # integer(0)
+        # Error in print(x) : object 'x' not found
+    }
+    
+    return (@arr > 1) ? @arr : $arr[0];
+}
+
+
+sub create_hash {
+    # creates a hash with the given properties
+    my (%property) = @_;
+    
+    # local variables
+    
+    # as defaults
+    # with no property          croak "property hash required" unless keys(%property) > 0;
+    
+    
+    # with the current package namespace
+    my $hash = {};
+    bless $hash;
+    
+    # creates a hash
+    foreach my $name (keys %property)
+    {
+        # assigns the property with its default
+        $hash->{"_$name"} = $property{$name};
+    }
+    
+    return $hash;
+}
+
+
+sub get_filename {
+    # param: path, return: file name and extension
+    # gets the file name
+    my $filepath = shift;
+
+    # On Unix returns ("baz", "/foo/bar", ".txt") from "/foo/bar/baz.txt"
+    my ($name, $path, $ext) = fileparse($filepath, qr/\.[^.]*/);
+    return $name . $ext;
+}
+
+
+sub is_numeric {
```

Comment:
The function ```looks_like_number``` in the module Scalar::Util does this same job and is more thorough.